### PR TITLE
Respond to #errors like an ActiveModel

### PR DESCRIPTION
**Pull request by ceslami**
@jmileham @samandmoore @reustmd @LucasSantoro

### CODE REVIEW

File: lib/her/model/active_model_overrides.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module Her
+  module Model
+    module ActiveModelOverrides
+      extend ActiveSupport::Concern
+
+      def errors
+        errors = super
+        errors.messages.merge! @response_errors unless @response_errors.empty?
```
Comment (jmileham) [inline lib/her/model/active_model_overrides.rb]: This seems destructive of a key already existing on messages, right? I'd like to see the response_error values append to the same key that already exists on the base errors if possible.  You'd then need to have an implementation that wouldn't keep appending the same messages each time you called errors.

File: spec/model/orm_spec.rb [hunk 2]
```diff
@@ -312,7 +312,7 @@ def friends
         builder.use Faraday::Request::UrlEncoded
         builder.adapter :test do |stub|
           stub.post("/users") { |env| [200, {}, { :id => 1, :fullname => Faraday::Utils.parse_query(env[:body])['fullname'], :email => Faraday::Utils.parse_query(env[:body])['email'] }.to_json] }
-          stub.post("/companies") { |env| [422, {}, { :errors => ["name is required"] }.to_json] }
+          stub.post("/companies") { |env| [422, {}, { :errors => { :name => ["name is required"] } }.to_json] }
```
Comment (jmileham) [inline spec/model/orm_spec.rb]: Typically a rails app would respond with `name: ['is required']`, if we're going for testing the behavior of an railsy API by default.

Comment (jmileham) [inline spec/model/orm_spec.rb]: Not sure if this was from the original specs or not, not a big deal either way.

**Comment by jmileham** [comment]
That's all I got for now, merging in errors on a per-value basis is going to be important though so we can have some local validations and some remote validations on the same attribute play nice together.

File: spec/model/orm_spec.rb (hunk 2 above)
Comment (samandmoore) [inline spec/model/orm_spec.rb]: yep. and then `full_messages` would give you the product of key with values => `name is required`

File: lib/her/model/active_model_overrides.rb (hunk 1 above)
Comment (ceslami) [inline lib/her/model/active_model_overrides.rb]: Talked offline -- it holds that if we had any local validation errors, they would need to be cleaned up before remote work is done. And at that point, there's no reason to protect the now-out-of-date validation errors.

**Comment by jmileham** [comment]
Talked offline, @ceslami correctly points out that while what we're going for is a sum of local and remote errors, we'll never go out to the server if there are local validation errors, so no need to fold-in.  I'd maybe add a test to make sure nobody regresses the behavior if you save the same model twice with remote validation errors (shouldn't get two copies of those errors - I think the current behavior should be fine for that).

**Comment by ceslami** [comment]
bümp

**Comment by jmileham** [comment]
LGTM

**Pull request merged**